### PR TITLE
Handle `context` cancellation in some of the `querier` downstream requests

### PR DESCRIPTION
**Pull request by kavirajk**
**What this PR does / why we need it**:
This PR handles `context` cancellation in some of the `querier` downstream requests.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by cyriltovena** [comment]
This include change we already merge or are in-review process. Can we just keep context check one we added ?

**Comment by kavirajk** [comment]
@cyriltovena Yes. done :)

File: pkg/querier/http.go [hunk 1]
```diff
@@ -34,6 +35,15 @@ type QueryResponse struct {
...
 func (q *Querier) RangeQueryHandler(w http.ResponseWriter, r *http.Request) {
+	log := spanlogger.FromContext(r.Context())
+	level.Info(log).Log("msg", "range query started")
+	defer func() {
+		level.Info(log).Log("msg", "range query ended")
+	}()
+	go func() {
+		<-r.Context().Done()
+		level.Info(log).Log("msg", "range query done", "err", r.Context().Err())
+	}()
```
Comment (cyriltovena) [inline pkg/querier/http.go:46]:
```suggestion
```

Let's remove this it was for debugging.

**Review by cyriltovena — APPROVED** [review]
LGTM

**Comment by cyriltovena** [comment]
You have a build error, my bad my suggestion didn't remove the import.

File: pkg/querier/http.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"time"
 
 	util_log "github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/cortexproject/cortex/pkg/util/spanlogger"
```
Comment (cyriltovena) [inline pkg/querier/http.go:9]: this one.

**Review by owen-d — APPROVED** [review]

**Pull request merged**